python/cudf/cudf/core/multiindex.py

```diff
@@ -108,10 +111,32 @@ def __init__(
             else:
                 level = DataFrame({name: self._levels[i]})
             level = DataFrame(index=codes).join(level)
-            self._source_data[name] = level[name].reset_index(drop=True)
+            source_data[name] = level[name].reset_index(drop=True)
 
+        self._data = source_data._data
         self.names = [None] * len(self._levels) if names is None else names
 
+    @classmethod
+    def _from_table(cls, table):
+        df = cudf.DataFrame(table._data)
+        return MultiIndex.from_frame(df, names=df.columns)
+
+    @property
+    def _source_data(self):
+        return cudf.DataFrame(self._data)
+
+    @_source_data.setter
+    def _source_data(self, value):
+        self._data = value._data
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
```

Comment:
Could be good to know if the name is actually being used anywhere for MultiIndex. If it's not let's throw it out.

Reply:
It is unfortunately. Tests fail catastrophically if I get rid of these and I'd rather tackle it in another PR.